server.go

```diff
@@ -138,8 +141,12 @@ func NewServer(options ...func(*Server)) *Server {
 		option(s)
 	}
 
+	addr := s.Addr
+	if s.listener != nil {
+		addr = s.listener.Addr().String()
+	}
```

Comment:
It is possible for the user to do something poor here say the user adds their own listener and then their own withaddr. The value could be off since listener trumps everything. I plan on getting rid of this possibility with #248. Basically pushing the openapi server setup as close to startup as we can.

Reply:
Good idea. Good thing mux and server are decorrelated in the std lib